Fix occasional build failure due to "too many open files" error

It happens sometimes during the RPM builds in presubmit workflows, even though we raise the open file limit to 64K [here](https://github.com/microshift-io/microshift/blob/main/.github/actions/build/action.yaml#L48).

```
vendor/k8s.io/kubernetes/pkg/kubelet/container/testing/mock_runtime_cache.go:26:2: open /var/home/microshift/microshift/_output/rpmbuild/BUILD/microshift-58c6812e489c031c6c66ff614d170eb45cd8394d/vendor/github.com/stretchr/testify/mock/doc.go: too many open files
make[2]: *** [/var/home/microshift/microshift/_output/rpmbuild/BUILD/microshift-58c6812e489c031c6c66ff614d170eb45cd8394d/vendor/github.com/openshift/build-machinery-go/make/targets/golang/build.mk:16: build] Error 1
```

--- a/.github/actions/build/action.yaml
+++ b/.github/actions/build/action.yaml
@@ -37,15 +37,19 @@ runs:
         sudo ln -s /mnt/tmp /var/tmp
         sudo ln -s /mnt/containers /var/lib/containers
 
+        # Print the current system limits
+        ulimit -a
+
+        # Raise open file limits to avoid "too many open files" errors
+        echo '* soft nofile 524288' | sudo tee -a /etc/security/limits.conf &>/dev/null
+        echo '* hard nofile 524288' | sudo tee -a /etc/security/limits.conf &>/dev/null
+
     - name: Build MicroShift RPMs
       if:  inputs.action == 'build-rpms' || inputs.action == 'build-all'
       shell: bash
       run: |
         # See https://github.com/microshift-io/microshift/blob/main/docs/build.md
         # for more information about the build process.
-
-        # Raise open file limits to avoid "too many open files" errors.
-        ulimit -n 65536
 
         # Run the RPM build process.
         cd ${GITHUB_WORKSPACE}/
@@ -60,9 +64,6 @@ runs:
       run: |
         # See https://github.com/microshift-io/microshift/blob/main/docs/build.md
         # for more information about the build process.
-
-        # Raise open file limits to avoid "too many open files" errors.
-        ulimit -n 65536
 
         # Run the container image build process.
         cd ${GITHUB_WORKSPACE}/

